File: app/src/main/java/com/v/vapp/activity/MainActivity.java
package com.v.vapp.activity;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.blankj.utilcode.util.ActivityUtils;
import com.v.vapp.DataHolder;
import com.v.vapp.R;
import com.v.vapp.databinding.ActivityMainBinding;
import com.v.vapp.viewModel.MainVM;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        ActivityUtils.finishAllActivities();
        super.onCreate(savedInstanceState);
        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        binding.setModel(new MainVM(this));
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        DataHolder.getInstance().logout();
    }
}

File: minesweeper/src/main/java/com/v/minesweeper/framework/activity/MainActivity.kt
package com.v.minesweeper.framework.activity

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import com.v.minesweeper.R

/**
 * 游戏主界面
 */
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    @Suppress("UNUSED_PARAMETER")
    fun startGame(v: View) {
        startActivity(Intent(this, MineActivity::class.java))
    }

    @Suppress("UNUSED_PARAMETER")
    fun settings(v: View) {
        startActivity(Intent(this, SettingsActivity::class.java))
    }
}
